src/ResourceManager/DevSpaces/Commands.DevSpaces/Commands/GetAzureRmDevSpacesController.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.DevSpaces.Models;
+using Microsoft.Azure.Commands.DevSpaces.Properties;
+using Microsoft.Azure.Commands.DevSpaces.Utils;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.DevSpaces.Commands
+{
+    [Cmdlet(VerbsCommon.Get, DevSpacesControllerNoun, DefaultParameterSetName = ListDevSpacesControllerParameterSet)]
+    [OutputType(typeof(PSController), typeof(IList<PSController>))]
+    public class GetAzureRmDevSpacesController : DevSpacesCmdletBase
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = false,
+            ParameterSetName = ListDevSpacesControllerParameterSet,
+            HelpMessage = "Resource group name")]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = DevSpacesControllerNameParameterSet,
+            HelpMessage = "Resource group name")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = DevSpacesControllerNameParameterSet,
+            HelpMessage = "DevSpaces controller name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The DevSpaces resource id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
```

Comment:
Because this is a top level resource, you don't need the InputObject/ResourceId parameter sets.  This is just required for Set/Remove/Update for top level resources.  In general, Get cmdlets pipe in their parent object, which in this case doesn't exist.

Reply:
Done